src/main/java/reposense/RepoSense.java

```diff
@@ -85,20 +85,29 @@ public static void main(String[] args) {
             RepoConfiguration.setIsFindingPreviousAuthorsPerformedToRepoConfigs(configs,
                     cliArguments.isFindingPreviousAuthorsPerformed());
 
-            if (RepoConfiguration.isAnyRepoFindingPreviousAuthors(configs)
-                    && !GitVersion.isGitVersionSufficientForFindingPreviousAuthors()) {
+            boolean isGitVersionInsufficient = RepoConfiguration.isAnyRepoFindingPreviousAuthors(configs)
+                    && !GitVersion.isGitVersionSufficientForFindingPreviousAuthors();
+            if (isGitVersionInsufficient) {
                 logger.warning(FINDING_PREVIOUS_AUTHORS_INVALID_VERSION_WARNING_MESSAGE);
                 RepoConfiguration.setToFalseIsFindingPreviousAuthorsPerformedToRepoConfigs(configs);
             }
 
+            boolean isTestMode = cliArguments.isTestMode();
+
+            if (isTestMode) {
+                assert !isGitVersionInsufficient;
+                AuthorConfiguration.setHasAuthorConfigFile(false);
+            }
```

Comment:
Sorry I think I missed this the last time round. For insufficient git version even on test mode, please log instead. 

I'm not too familiar with exactly how the original system test cases were run. Could you explain the need for `AuthorConfiguration.setHasAuthorConfigFile(false);` and why it needs the git version check

Reply:
The assert is because the original behavior is to exit the test straight away when git is insufficient. I used assert as a temporary measure. Do you think I should make it throw an exception or just remove it?

As for the latter line, it is the only difference between the original version and the one in main. Removing this will cause some test cases to fail. As I'm not familiar with how the old test cases works, I have kept this line for now.

Reply:
Noted on the second point. Would it be possible to set it directly in the `runTest`method in ConfigSystemTest?

Also i misremembered the original behaviour but to clarify, it aborts the test instead of outright failing which affects whether someone who has a lower git version is able to properly build. 

I think it might be possible to do this in runTest by checking if the inputbuilder has the finding previous author flag and then doing `Assumptions.assumeFalse` similar to the original

Reply:
For the first, I doubt it can be achieved. It involves setting some parameters that is within main, which is against what this PR is achieving.

Another way I can think of is probably to add a static boolean variable in main and let test cases manipulate that. Then use the variable to set this line. Some what similar to what was currently done but without the need to use a flag.